### ISSUE TITLE
Using Dataloader in pytorch_lightning when using DDP training?

### ISSUE THREAD

**Issue by NguyenDuyDuc1491995** [Lightning-AI/lightning#13360]
My computer has 2 gpus and I have some problems:
1. First, I used Pytorch
I have 10 images, I created distributed dataloader (using sampler) follow Pytorch instruction, batchsize = 4, gpu=2.
=> So with each gpu, length of batch0 is 4 and length of batch1 is 4. Sampler added 2 to batch1 to make batchsize = 4.
I trained with ddp_spawn with pytorch code and everything is ok.

2. Next, I used Pytorch Lightning
I also use 10 images, I created dataloader (Normal Dataloader) follow Pytorch Instruction, batchsize =4, gpu=2
=> so with each gpu, length of batch0 is 4 and length of batch1 is 1.

Now, I want to use pytorch lightning but I want batchsize=4 same distributed sampler when working with pytorchlightning. How should I do?

Thanks

cc @awaelchli @rohitgr7 @akihironitta @justusschock @ninginthecloud @otaj

**Comment by NguyenDuyDuc1491995** [comment]
The problem is when I train semantic segmentation DeepLabV3 by pytorchlightning
Batch0 =4 is ok, but batch1=1 is error. Because the batch is 1 so there is a problem with batchnorm.

But I used distributed dataloader with sampler in pytorch and I saw it will create batch0=4 and batch1=4 ( it will take 3 more images) so I wonder if I can create a distributed dataloader as same as above in pytorch lightning?

**Comment by awaelchli** [comment]
The distributed sampler has nothing to do with batching. It only ensures that each GPU gets the same amount of samples and since your dataset of 10 images is divisible by 2, that's never a problem. 

To avoid uneven batch sizes, just set `drop_last=True` in the dataloader and then you are guaranteed to get the batch size 4 for each batch. This is the same in PyTorch and PL, there should be no difference.

**Comment by NguyenDuyDuc1491995** [comment]
ok, thanks

**Comment by awaelchli** [comment]
I hope the question was answered and the issues could be solved. Let us know if there is anything else causing trouble.
